sdk/python/tests/test_historical_retrieval.py

```diff
@@ -557,3 +605,74 @@ def test_historical_features_from_bigquery_sources(
             .reset_index(drop=True),
             check_dtype=False,
         )
+
+
+@pytest.mark.integration
+def test_feature_name_collision_on_historical_retrieval_from_parquet_sources():
+
+    driver_source = FileSource(
+        path="driver_stats_path",
+        event_timestamp_column="datetime",
+        created_timestamp_column="created",
+    )
+    driver_fv = create_driver_hourly_stats_feature_view(driver_source)
+    customer_source = FileSource(
+        path="customer_profile_path",
+        event_timestamp_column="datetime",
+        created_timestamp_column="created",
+    )
+    customer_fv = create_customer_daily_profile_feature_view(customer_source)
+
+    # _validate_and_group_feature_refs is the function that checks for colliding feature names
+    with pytest.raises(FeatureNameCollisionError):
+        _validate_and_group_feature_refs(
+            feature_refs=[
+                "driver_stats:conv_rate",
+                "driver_stats:avg_daily_trips",
+                "customer_profile:current_balance",
+                "customer_profile:avg_passenger_count",
+                "customer_profile:lifetime_trip_count",
+                "customer_profile:avg_daily_trips",
+            ],
+            all_feature_views=[driver_fv, customer_fv],
+            full_feature_names=False,
+        )
+
+
+def test_feature_name_collision_on_historical_retrieval_from_bigquery_sources():
```

Comment:
This logic isn't specific to data sources, so I wouldn't add this test case unnecessarily